http_logs/index.json

```diff
@@ -12,38 +12,118 @@
-        "type": "keyword",
-        "index": false,
-        "doc_values": false
-      },
-      "request": {
-        "type": "text",
-        "fields": {
-          "raw": {
-            "ignore_above": 256,
-            "type": "keyword"
+      {%- if runtime_grok is defined %}
+        "message": {
+          "type": "wildcard",
+          "index": true,
+          "doc_values": true
+        },
+        "extracted": {
+          "type": "runtime",
+          "grok": {
+            "field": "message",
+            "pattern": "%{IPORHOST:clientip} %{HTTPDUSER} %{USER} \\[%{TIMESTAMP_ISO8601}\\] \"(?:%{WORD} %{NOTSPACE:request}(?: HTTP/%{NUMBER})?|%{DATA})\" %{NUMBER:status:long} (?:%{NUMBER:size:long}|-)"
           }
+        },
+        "clientip": {
+          "type": "alias",
+          "path": "extracted.clientip"
+        },
+        "request": {
+          "type": "alias",
+          "path": "extracted.request"
+        },
+        "status": {
+          "type": "alias",
+          "path": "extracted.status"
+        },
+        "size": {
+          "type": "alias",
+          "path": "extracted.size"
         }
-      },
-      "status": {
-        "type": "integer"
-      },
-      "size": {
-        "type": "integer"
-      },
-      "geoip" : {
-        "properties" : {
-           "country_name": { "type": "keyword" },
-           "city_name": { "type": "keyword" },
-           "location" : { "type" : "geo_point" }
+      {%- elif runtime_script_grok is defined %}
+        "message": {
+          "type": "keyword",
+          "index": false,
+          "doc_values": true
+        },
+        "clientip": {
+          "type": "runtime",
+          "runtime_type": "ip",
+          "script": "String m = doc[\"message\"].value; int end = m.indexOf(\" \"); emit(m.substring(0, end));"
+        },
+        "request": {
+          "type": "runtime",
+          "runtime_type": "keyword", {# This should be text with a subtype of keyword #}
+          "script": "String m = doc[\"message\"].value; int start = m.indexOf(\"\\\"\") + 1; int end = m.indexOf(\"\\\"\", start); emit(m.substring(start, end));"
+        },
+        "status": {
+          "type": "runtime",
+          "runtime_type": "long",
+          "script": "String m = doc[\"message\"].value; int end = m.lastIndexOf(\" \"); int start = m.lastIndexOf(\" \", end - 1) + 1; emit(Long.parseLong(m.substring(start, end)));"
+        },
+        "size": {
+          "type": "runtime",
+          "runtime_type": "long",
+          "script": "String m = doc[\"message\"].value; int start = m.lastIndexOf(\" \") + 1; emit(Long.parseLong(m.substring(start)));"
+        }
+      {%- else %}
+        "message": {
+          "type": "keyword",
+          "index": false,
+          "doc_values": false
+        },
+        "clientip": {
+          "type": "ip",
+          "doc_values": {{ doc_values | default(true) | tojson }},
+          "index": {{ index | default(true) | tojson }}
```

Comment:
Except that I've added all of these so you can turn off indexing or doc values if you need to get a size estimate.